src/wfs/src/main/java/org/geoserver/wfs/response/CSVOutputFormat.java

```diff
@@ -384,4 +385,26 @@ public String getCapabilitiesElementName() {
     public String getCharset(Operation operation) {
         return gs.getGlobal().getSettings().getCharset();
     }
+
+    /**
+     * Checks if the used namespace prefix is available on GeoServer namespaces, and replace the
+     * namespace URI with the prefix name found.
+     *
+     * @param attributeName the current attribute name
+     * @return the fixed prefixed name, of the original attribute name if no namespace is found
+     */
+    String resolveNamespacePrefixName(String attributeName) {
+        if (StringUtils.isBlank(attributeName) || !attributeName.contains(":")) {
+            return attributeName;
+        }
+        int lastIndexOfSeparator = attributeName.lastIndexOf(":");
+        String namespaceUri = attributeName.substring(0, lastIndexOfSeparator);
+        NamespaceInfo namespace = this.gs.getCatalog().getNamespaceByURI(namespaceUri);
+        if (namespace != null) {
+            String localName =
+                    attributeName.substring(lastIndexOfSeparator + 1, attributeName.length());
```

Comment:
Should not happen but... what if someone silly put a ":" at the end of the attribute name?

Reply:
Indeed, good point.  We already allow ":" characters in the middle to support URIs definitions (excluding the last one taking it as namespace separator), so any attribute using it as final character should not be supported by this resolver method.  Applying an extra validation condition for this, thanks!